gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/OptionParsers.java

```diff
@@ -0,0 +1,258 @@
+ */
+class OptionParsers {
+
+  static Map<String, Object> parseListZonesOptions(String query) {
+    Map<String, Object> options = new HashMap<>();
+    if (query != null) {
+      String[] args = query.split("&");
+      for (String arg : args) {
+        String[] argEntry = arg.split("=");
+        switch (argEntry[0]) {
+          case "fields":
+            // List fields are in the form "managedZones(field1, field2, ...),nextPageToken"
+            String replaced = argEntry[1].replace("managedZones(", ",");
+            replaced = replaced.replace(")", ",");
+            // we will get empty strings, but it does not matter, they will be ignored
+            options.put("fields", replaced.split(","));
+            break;
+          case "dnsName":
+            options.put("dnsName", argEntry[1]);
+            break;
+          case "pageToken":
+            options.put("pageToken", argEntry[1]);
+            break;
+          case "maxResults":
+            // parsing to int is done while handling
+            options.put("maxResults", argEntry[1]);
+            break;
+          default:
+            break;
+        }
+      }
+    }
+    return options;
+  }
+
+  static String[] parseGetOptions(String query) {
+    if (query != null) {
+      String[] args = query.split("&");
+      for (String arg : args) {
+        String[] argEntry = arg.split("=");
+        if (argEntry[0].equals("fields")) {
+          // List fields are in the form "fields=field1, field2,..."
+          return argEntry[1].split(",");
+        }
+      }
+    }
+    return null;
+  }
+
+  static ManagedZone extractFields(ManagedZone fullZone, String[] fields) {
+    if (fields == null) {
+      return fullZone;
+    }
+    ManagedZone managedZone = new ManagedZone();
+    for (String field : fields) {
+      switch (field) {
+        case "creationTime":
+          managedZone.setCreationTime(fullZone.getCreationTime());
+          break;
+        case "description":
+          managedZone.setDescription(fullZone.getDescription());
+          break;
+        case "dnsName":
+          managedZone.setDnsName(fullZone.getDnsName());
+          break;
+        case "id":
+          managedZone.setId(fullZone.getId());
+          break;
+        case "name":
+          managedZone.setName(fullZone.getName());
+          break;
+        case "nameServerSet":
+          managedZone.setNameServerSet(fullZone.getNameServerSet());
+          break;
+        case "nameServers":
+          managedZone.setNameServers(fullZone.getNameServers());
+          break;
+        default:
+          break;
+      }
+    }
+    return managedZone;
+  }
+
+  static Change extractFields(Change fullChange, String[] fields) {
+    if (fields == null) {
+      return fullChange;
+    }
+    Change change = new Change();
+    for (String field : fields) {
+      switch (field) {
+        case "additions":
+          // todo the fragmentation is ignored here as our api does not support it
```

Comment:
Is fragmentation something plan to support?  If so, it'd be nice to make this TODO reference an issue number (here and below)